app/controllers/posts_controller.rb

```diff
@@ -9,7 +9,8 @@ def new
   end
 
   def create
-    @post = Post.create(post_params)
+    @user = User.find(current_user.id)
+    @user.posts.create(post_params)
```

Comment:
You are redirecting, so you don't need the instance variable here - it is never being called in a template.